### PR TITLE
feat: Get active span

### PR DESCRIPTION
**Pull request by ujwaldhakal**
Added a method to return the active span so that one can have flexibility to hook other process.

The current Symfony bundle is not compatible with Symfony Messenger. So I had to hook with these [messenger events](https://symfony.com/doc/current/messenger.html#messenger-events) for this I need tracer object to manually close & flush if there are any active connections

### CODE REVIEW

**Comment by ujwaldhakal** [comment]
@cawolf the functional tests are not running on my local every time I run `./vendor/bin/phpunit --testsuite functional` I get
`Symfony\Component\Process\Exception\RuntimeException: The provided cwd "build/testproject" does not exist.`. And I couldn't see any docs to run functional tests,  am I missing anything?

**Review by cawolf — CHANGES_REQUESTED** [review]
Yes, the docs are still missing, usually there are no problems with the functional tests. However, in the `composer.json` there is a script block called `phpunit-functional`. To be able to run this locally, you usually need
* to have `docker `set up
* to have `symfony` CLI set up
* set an environment variable like `export SYMFONY_VERSION=4.4.*`
* set an environment variable like `export GITHUB_HEAD_REF=refs/heads/dev-get-active-span`
* run the tests

But you are now the first to run the actions on a fork of this project, and the steps laid out only work for pull requests of the project itself. What it does internally is installing the current working branch through composer, but the repository source is pointing to the original repository, not your fork. We need to replace this during the setup of your functional tests, so please add the code snippet to the Tests/Functional/Scripts/requireAdditionalVendors.sh in line 11:
```
    composer config repositories.fork vcs ${GITHUB_SERVER_URL}/${GITHUB_REPOSITORY}.git
```

With this committed, you now need to additionally:
* set an environment variable like `export GITHUB_SERVER_URL=https://github.com`
* set an environment variable like `export GITHUB_REPOSITORY=ujwaldhakal/OpentracingBundle-core`
* run the tests with `composer phpunit-functional`

**Comment by ujwaldhakal** [comment]
Thanks, @cawolf I am able to run the functional tests locally.

But the pipeline is taking quite long to get executed, I trigged this yesterday but its still pending. It says it need some approval

File: Tests/Functional/Scripts/requireAdditionalVendors.sh
Comment (cawolf) [inline Tests/Functional/Scripts/requireAdditionalVendors.sh:11]: This really needs to be the environment variable names I proposed, because they will be set automatically in GitHub actions:
```suggestion
    composer config repositories.fork vcs ${GITHUB_SERVER_URL}/${GITHUB_REPOSITORY}.git
```

**Review by cawolf — CHANGES_REQUESTED** [review]
The approval seems not to be saved for you, I already did that some time ago.

Anyways, I added a comment for the additional vendors script, and could you please revert the changes to `doc/img/esi-trace.png`? Thank you

File: Tests/Functional/Scripts/requireAdditionalVendors.sh
Comment (ujwaldhakal) [reply Tests/Functional/Scripts/requireAdditionalVendors.sh]: Thank you, I am quite stucked with the breaklines lines for the image here

I tried to revert by using `git checkout 2cbf03e -- doc/img/esi-trace.png`
```
warning: CRLF will be replaced by LF in doc/img/esi-trace.png.
The file will have its original line endings in your working directory
```
With this warning and when I try to add them my github automatically reverted to my current state with no changes . How do you come with this kind of issue specially when there is multiple os involved?

**Review by cawolf — CHANGES_REQUESTED** [review]
There is still one thing missing to merge this: you need to configure a `GH_TOKEN` actions variable in your fork, using a read-only personal access token. Otherwise, the checkout will not work properly for the functionals.

Regarding the line ending issue in the image: you could check out the `master` with the correct line ending option, save the file somewhere else and copy it into your branch.

**Comment by ujwaldhakal** [comment]
> GH_TOKEN



> There is still one thing missing to merge this: you need to configure a `GH_TOKEN` actions variable in your fork, using a read-only personal access token. Otherwise, the checkout will not work properly for the functions.
> 
> Regarding the line ending issue in the image: you could check out the `master` with the correct line ending option, save the file somewhere else, and copy it into your branch.

I have added secrets on my forked repository with the name `GH_TOKEN`
And I have reverted the image file, the automatic conversion was due to the `* text eol=lf` inside` .gitattributes`

**Comment by cawolf** [comment]
I'm currently debugging the functional tests for forks with a second account, mimicking your PR. I will not merge it, but I need to see in depth what is happening and why the functionals are failling.

**Comment by cawolf** [comment]
I fixed the functional test issues. As soon as the last report is coming back, I can merge your PR.

**Comment by ujwaldhakal** [comment]
> I fixed the functional test issues. As soon as the last report is coming back, I can merge your PR.

Thank you. Really appreciate for fixing this issue. Now we can merge

**Review by cawolf — APPROVED** [review]

**Pull request merged**
